### PR TITLE
SITES-685 improve news page performance

### PR DESCRIPTION
**Pull request by freshtonic**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ This file is influenced by http://keepachangelog.com/.
 ### Fixed
 - Pluralise 'ministers' only where appropriate
 - Fixed HSTS settings so we don't force every gov.au subdomain to run over https
+- Perforance of /news page
```
Comment (ltankey) [inline CHANGELOG.md]: Typo: just missing an `m` 😃

Comment (freshtonic) [inline CHANGELOG.md]: LOL

**Comment by bjgordon** [comment]
Nice, LGTM

File: app/helpers/news_helper.rb [hunk 2]
```diff
@@ -8,10 +8,18 @@ def news_article_class
...
-    else
-      NewsArticle.published.by_release_date.by_published_at
-    end
+    query_root = if section.present?
+                   section.news_articles
+                 else
+                   NewsArticle
+                 end
+    query_root
+      .preload([
```
Comment (coderdan) [inline app/helpers/news_helper.rb]: I hadn't actually seen preload. Is it new in Rails 5?

File: app/helpers/news_helper.rb [hunk 3]
```diff
@@ -8,10 +8,18 @@ def news_article_class
   end
 
   def news_articles(section: nil)
-    if section.present?
-      section.news_articles.published.by_release_date.by_published_at
-    else
-      NewsArticle.published.by_release_date.by_published_at
-    end
+    query_root = if section.present?
```
Comment (coderdan) [inline app/helpers/news_helper.rb]: Could you make query_root a method to keep this method a bit simpler?

File: app/helpers/news_helper.rb (hunk 2 above)
Comment (freshtonic) [inline app/helpers/news_helper.rb]: I only just came across it. AFAIK it does exactly the same as `includes` but I think the name is more intention-revealing. From the docs:

File: app/helpers/news_helper.rb (hunk 3 above)
Comment (freshtonic) [inline app/helpers/news_helper.rb]: Done. See latest commit.

**Comment by ltankey** [comment]
LGTM

File: app/helpers/news_helper.rb [hunk 4]
```diff
@@ -7,11 +7,20 @@ def news_article_class
...
+  def preload_news_articles(query_root = NewsArticle)
+    query_root
+      .preload([
+        {:section  => {:home_node => :parent}},
+        {:sections => {:home_node => :parent}}
+      ])
+      .published
+      .by_release_date
+      .by_published_at
+      .by_name
```
Comment (ltankey) [inline app/helpers/news_helper.rb]: is by_name necessary? Published_at is a timestamp - is this just for edge cases?

File: app/controllers/news_controller.rb [hunk 5]
```diff
@@ -13,9 +11,9 @@ def index
     if params[:section].present?
       set_section
       set_menu_nodes
-      @articles = news_articles section: @section
+      @articles = preload_news_articles(@section.news_articles)
```
Comment (ltankey) [inline app/controllers/news_controller.rb]: Do we need to pass `@section`? If the helper is included, it would have access to it by default.

**Comment by ltankey** [comment]
minor comments aside, LGTM

File: app/helpers/news_helper.rb (hunk 4 above)
Comment (freshtonic) [inline app/helpers/news_helper.rb]: There was an existing spec that tests for order by dates then name. The previous implementation did not sort by name and it was accidentally passing. So I made it sort by name.

**Pull request merged**